pkg/gen/node-templates/provider.ts.mustache

```diff
@@ -379,13 +391,13 @@ export namespace {{Group}} {
       * @param args The arguments to use to populate this resource's properties.
       * @param opts A bag of options that control this resource's behavior.
       */
-      constructor(name: string, args: inputApi.{{Group}}.{{Version}}.{{Kind}}, opts?: pulumi.CustomResourceOptions) {
+      constructor(name: string, args?: inputApi.{{Group}}.{{Version}}.{{Kind}}, opts?: pulumi.CustomResourceOptions) {
```

Comment:
Why do we need this?

Reply:
All the other packages do it this way and it's annoying to have to manually specify an empty object when all the properties are optional.

Reply:
I believe we only do this for resources which have all optional input argument object properties.  I'm less sure this makes sense to make optional overall.

Reply:
@lukehoban `get` needs to call the constructor with `null` as args. I'm not sure of a really great way to make this work. One option is to do what we're doing here, which doesn't seem so bad because omitting it will result in a validation error anyway.

Reply:
If this is how we do it in TF implementations then fine to kee it the same here. I recall thinking the approach there was awkward too - but better to be consistent.